src/web/app/pages-instructor/instructor-course-enroll-page/instructor-course-enroll-page.component.ts

```diff
@@ -94,59 +106,211 @@ export class InstructorCourseEnrollPageComponent implements OnInit {
+        this.allStudentChunks.push(currentStudentChunk);
+        currentStudentChunk = [];
+      }
+    }
+    if (currentStudentChunk.length > 0) {
+      this.allStudentChunks.push(currentStudentChunk);
+    }
+  }
+
+  private checkTeamsValid(studentEnrollRequests: Map<number, StudentEnrollRequest>): void {
+    const teamSectionMap: Map<string, string> = new Map();
+    const teamIndexMap: Map<string, number> = new Map();
+    const noInvalidRowsBefore: number = this.invalidRowsIndex.size;
+
+    Array.from(studentEnrollRequests.keys()).forEach((key: number) => {
+      const request: StudentEnrollRequest | undefined = studentEnrollRequests.get(key);
+      if (request === undefined) {
+        return;
+      }
+
+      if (teamSectionMap.has(request.team)) {
+        if (teamSectionMap.get(request.team) !== request.section) {
+          this.invalidRowsIndex.add(key);
+          const firstIndex: number | undefined = teamIndexMap.get(request.team);
+          if (firstIndex !== undefined) {
+            this.invalidRowsIndex.add(firstIndex);
+          }
         }
-        this.isExistingStudentsPresent = true;
-      });
+      } else {
+        teamSectionMap.set(request.team, request.section);
+        teamIndexMap.set(request.team, key);
+      }
+    });
+    if (this.invalidRowsIndex.size > noInvalidRowsBefore) {
+      this.enrollErrorMessage += 'Found duplicated teams in different sections. ';
+    }
+  }
+
+  private checkCompulsoryFields(studentEnrollRequests: Map<number, StudentEnrollRequest>): void {
+    const noInvalidRowsBefore: number = this.invalidRowsIndex.size;
+
+    Array.from(studentEnrollRequests.keys()).forEach((key: number) => {
+      const request: StudentEnrollRequest | undefined = studentEnrollRequests.get(key);
+      if (request === undefined) {
+        return;
+      }
+
+      if ((studentEnrollRequests.size >= 100 && request.section === '')
+          || request.team === '' || request.name === '' || request.email === '') {
+        this.invalidRowsIndex.add(key);
+      }
+    });
+    if (this.invalidRowsIndex.size > noInvalidRowsBefore) {
+      this.enrollErrorMessage += 'Found empty compulsory fields. ';
+    }
+  }
+
+  private checkEmailNotRepeated(studentEnrollRequests: Map<number, StudentEnrollRequest>): void {
+    const emailMap: Map<string, number> = new Map();
+    const noInvalidRowsBefore: number = this.invalidRowsIndex.size;
+
+    Array.from(studentEnrollRequests.keys()).forEach((key: number) => {
+      const request: StudentEnrollRequest | undefined = studentEnrollRequests.get(key);
+      if (request === undefined) {
+        return;
+      }
+
+      if (emailMap.has(request.email)) {
+        this.invalidRowsIndex.add(key);
+        const firstIndex: number | undefined = emailMap.get(request.email);
+        if (firstIndex !== undefined) {
+          this.invalidRowsIndex.add(firstIndex);
+        }
+      } else {
+        emailMap.set(request.email, key);
```

Comment:
ditto; by first setting it, the happy path does not need to be in if-else block as the element should exist.